tool/tctl/common/device/resource.go

```diff
@@ -0,0 +1,161 @@
+// Copyright 2023 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package device
+
+import (
+	"time"
+
+	"github.com/gravitational/trace"
+	log "github.com/sirupsen/logrus"
+	"google.golang.org/protobuf/types/known/timestamppb"
+
+	devicepb "github.com/gravitational/teleport/api/gen/proto/go/teleport/devicetrust/v1"
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/lib/devicetrust"
+	"github.com/gravitational/teleport/lib/utils"
+)
+
+// Resource is a wrapper around devicepb.Device that implements types.Resource.
+type Resource struct {
+	// ResourceHeader is embedded to implement types.Resource.
+	types.ResourceHeader
+	// Spec is the device specification.
+	Spec ResourceSpec `json:"spec"`
+}
+
+// ResourceSpec is the device resource specification.
+// This spec is intended to closely mirror [devicepb.Device] but swaps some data around
+// to get a UX that matches with our other resource types.
+type ResourceSpec struct {
+	// OsType is represented as a string here for user-friendly manipulation.
+	OsType      string                      `json:"os_type"`
```

Comment:
This probably should have been `OSType`: https://github.com/golang/go/wiki/CodeReviewComments#initialisms

Reply:
Maybe, I followed this due to the naming in the existing code in https://github.com/gravitational/teleport/blob/master/api/gen/proto/go/teleport/devicetrust/v1/device.pb.go#L107 generated by our protobuf spec.

Reply:
+1 for OSType, proto idiosyncrasies shouldn't influence pure Go code. I can follow up later, there are some PRs I need to write around this